cond.go

```diff
@@ -0,0 +1,253 @@
+	}
+
+	return " ="
+}
+
+func compileLeaf(leaf interface{}) (string, []interface{}, error) {
+	if _, ok := leaf.(string); ok {
+		return leaf.(string), nil, nil
+	}
+
+	if sqlizer, ok := leaf.(squirrel.Sqlizer); ok {
+		return sqlizer.ToSql()
+	}
+
+	return "?", []interface{}{leaf}, nil
+}
+
+// Cond is a map of conditions.
+type Cond map[interface{}]interface{}
+
+func (c Cond) ToSql() (string, []interface{}, error) {
+	nodes := []squirrel.Sqlizer{}
+
+	for left, right := range c {
+		nodes = append(nodes, &condNode{left, right})
+	}
+
+	return compileNodes(nodes)
+}
+
+// Eq represents an equality comparison.
+func Eq(v interface{}) squirrel.Sqlizer {
+	return &condExpr{"=", v}
+}
+
+// Lt represents a less-than comparison.
+func Lt(v interface{}) squirrel.Sqlizer {
+	return &condExpr{"<", v}
+}
+
+// Lte represents a less-than-or-equal comparison.
+func Lte(v interface{}) squirrel.Sqlizer {
+	return &condExpr{"<=", v}
+}
+
+// IsNull represents an IS NULL comparison.
+func IsNull() squirrel.Sqlizer {
+	return &condExpr{"IS NULL", nil}
+}
+
+// IsNotNull represents an IS NOT NULL comparison.
+func IsNotNull() squirrel.Sqlizer {
+	return &condExpr{"IS NOT NULL", nil}
+}
+
+// Gt represents a greater-than comparison.
+func Gt(v interface{}) squirrel.Sqlizer {
+	return &condExpr{">", v}
+}
+
+// Gte represents a greater-than-or-equal comparison.
+func Gte(v interface{}) squirrel.Sqlizer {
+	return &condExpr{">=", v}
+}
+
+// NotEq represents a not-equal comparison.
+func NotEq(v interface{}) squirrel.Sqlizer {
+	return &condExpr{"<>", v}
+}
+
+// Like represents a LIKE comparison.
+func Like(v interface{}) squirrel.Sqlizer {
+	return &condExpr{"LIKE", v}
+}
+
+// ILike represents a ILIKE comparison.
+func ILike(v interface{}) squirrel.Sqlizer {
+	return &condExpr{"ILIKE", v}
+}
+
+// NotILike represents a NOT ILIKE comparison.
+func NotILike(v interface{}) squirrel.Sqlizer {
+	return &condExpr{"NOT ILIKE", v}
+}
+
+// NotLike represents a NOT LIKE comparison.
+func NotLike(v interface{}) squirrel.Sqlizer {
+	return &condExpr{"NOT LIKE", v}
+}
+
+// In represents an IN operator. The value must be variadic.
+func In(v ...interface{}) squirrel.Sqlizer {
+	return sqlExprFn(func() (string, []interface{}, error) {
+		if len(v) == 0 {
+			return "IN ()", nil, nil
+		}
+
+		return "IN (?" + strings.Repeat(", ?", len(v)-1) + ")", v, nil
+	})
+}
+
+// NotIn represents a NOT IN operator. The value must be variadic.
+func NotIn(v ...interface{}) squirrel.Sqlizer {
+	return sqlExprFn(func() (string, []interface{}, error) {
+		if len(v) == 0 {
+			return "NOT IN ()", nil, nil
+		}
+
+		return "NOT IN (?" + strings.Repeat(", ?", len(v)-1) + ")", v, nil
+	})
+}
+
+// AnyOf represents an IN operator. The value must be a slice.
+func AnyOf(v interface{}) squirrel.Sqlizer {
+	return sqlExprFn(func() (string, []interface{}, error) {
+		rv := reflect.ValueOf(v)
+
+		if rv.Kind() != reflect.Slice {
+			return "", nil, fmt.Errorf("value must be a slice")
+		}
```

Comment:
Just a thought, maybe this could also take array.
```suggestion
// AnyOf represents an IN operator. The value must be a slice or an array.
func AnyOf(v interface{}) squirrel.Sqlizer {
	return sqlExprFn(func() (string, []interface{}, error) {
		rv := reflect.ValueOf(v)

		rk := rv.Kind()
		if rk != reflect.Slice && rk != reflect.Array {
			return "", nil, fmt.Errorf("value must be a slice or an array")
		}
```

Reply:
what about `AnyOf[T](v []T)`?